extensions/adobe/experience/decisioning/frequency-capping-constraints.schema.json

```diff
@@ -55,45 +55,83 @@
                 "type": "string",
                 "format": "date-time",
                 "title": "End Date and Time",
-                "description": "Capping end date and time. Property has the semantic of schema.org's 'endTime' property defined on http://schema.org/Action"
+                "description": "Capping end date and time. Property has the semantic of schema.org's 'endTime' property defined on http://schema.org/Action",
+                "meta:status": "deprecated"
               },
               "xdm:duration": {
                 "title": "Duration",
                 "description": "Duration in hours. When xdm:repeat is true, daily scan job of the rules will update all expired rules to start from current time and end with the specific xdm:duration",
                 "type": "integer",
-                "minimum": 1
+                "minimum": 1,
+                "meta:status": "deprecated"
               },
               "xdm:events": {
                 "title": "Events",
-                "description": "Events of the capping to comply with. Since most of the events don't support offerId level counting, we will start with only support decisioning event, and will extend to support custom event in future phases. Other events to be supported in the future : web-display, web-click, message-delivery, message-open, message-click",
+                "description": "Events of the capping to comply with. Since most of the events don't support offerId level counting, we will start with only support decisioning event, and will extend to support custom event in future phases. Other events to be supported in the future : web-display, web-click, message-delivery, message-open, message-click.",
                 "type": "array",
                 "items": {
                   "type": "string",
-                  "enum": ["decisioning"],
+                  "enum": ["decisioning", "custom", "click", "impression"],
                   "meta:enum": {
-                    "decisioning": "Decisioning"
+                    "decisioning": "Decisioning",
+                    "custom": "Custom",
+                    "click": "Offer Click",
+                    "impression": "Offer Impression"
                   }
                 },
                 "minItems": 1,
                 "default": ["decisioning"]
               },
+              "xdm:customQuery": {
+                "title": "Custom PQL query",
+                "description": "Custom PQL query to match events.",
+                "type": "string"
+              },
               "xdm:repeat": {
                 "title": "Repeat",
-                "description": "When true, new counter rule with updated time window will be created. Configure a capping to be repeated (e.g., max times every 1 week). ECS will support repeating config in the rules, scan rules daily, and create a new rule with new expiry when the rule is due for repeating",
-                "type": "boolean",
```

Comment:
Changing the type of a property is also a breaking change. If you want to change "repeat" from "boolean" to an "object", you will have to deprecate the existing "repeat" property and create another one with a different name as an object. Thanks.

Reply:
@mjasrotia there is no data created with `repeat` boolean value.  this filed was added before repeat feature implementation and after architectural design changes we decided to have the `repeat` as json object instead of boolean, Now the API changes with repeat as json object in Production and we have data with `repeat` json object.